processors/filterprocessor/filter_processor.go

```diff
@@ -0,0 +1,305 @@
+package filterprocessor
+
+import (
+	"context"
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"reflect"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/indexer/data"
+	"github.com/algorand/indexer/plugins"
+	"github.com/algorand/indexer/processors"
+)
+
+const implementationName = "filter_processor"
+
+// package-wide init function
+func init() {
+	processors.RegisterProcessor(implementationName, &Constructor{})
+}
+
+// Constructor is the ProcessorConstructor implementation for the "address_filter" processor
+type Constructor struct{}
+
+// New initializes a blockProcessorConstructor
+func (c *Constructor) New() processors.Processor {
+	return &FilterProcessor{}
+}
+
+type expression interface {
+	Search(input interface{}) bool
+}
+
+type regexExpression struct {
+	Regex *regexp.Regexp
+}
+
+func (e regexExpression) Search(input interface{}) bool {
+	return e.Regex.MatchString(input.(string))
+}
+
+const constantExpressionStr = "const"
+const regexExpressionStr = "regex"
+
+func makeExpression(expressionType string, expressionSearchStr string) (*expression, error) {
+	switch expressionType {
+	case constantExpressionStr:
+		{
+			r, err := regexp.Compile("^" + expressionSearchStr + "$")
+			if err != nil {
+				return nil, err
+			}
+
+			var exp expression = regexExpression{Regex: r}
+			return &exp, nil
+		}
+	case regexExpressionStr:
+		{
+			r, err := regexp.Compile(expressionSearchStr)
+			if err != nil {
+				return nil, err
+			}
+
+			var exp expression = regexExpression{Regex: r}
+			return &exp, nil
+		}
+	default:
+		return nil, fmt.Errorf("unknown expression type: %s", expressionType)
+	}
+}
+
+type fieldOperation string
+
+const someFieldOperation fieldOperation = "some"
+const allFieldOperation fieldOperation = "all"
+
+type fieldSearcher struct {
+	Exp          *expression
+	Tag          string
+	MethodToCall string
+}
```

Comment:
I feel like moving `fieldSearcher` and `Expression` and all of their related methods to a separate module or package would make it easier to see what the expected usage of these is. As is, some of the methods are exported and some or not so it's a bit tough to figure out how this library of functions might be used by another package effectively.

It will also probably make testing them easier.

Reply:
Done